bin/run.sh

```diff
@@ -1,55 +1,89 @@
 #!/usr/bin/env bash
 
+shopt -s extglob
+
 usage() {
     printf '%s\n' "$*"
-    printf '%s\n' "Usage: ${0##*/} slug inputDir outputDir"
+    printf '%s\n' "Usage: ${0##*/} slug input_dir output_dir"
     exit 1
 }
 
-analyze() {
-    local opts=(
-        "--shell=bash"  
-        "--external-sources"
-        "--check-sourced"
-        "--norc"
-        "--format=json1"
-    )
-    (
-        cd "$inDir" >/dev/null || exit 1
-        shellcheck "${opts[@]}" "${filename}.sh"
-    )
+validate_input() {
+    if ! [[ -d "$in_dir" && -r "$in_dir" ]]; then
+        usage "Error: not a dir or not readable: ${in_dir@Q}"
+    fi
+    if ! [[ -d "$out_dir" && -w "$out_dir" ]]; then
+        usage "Error: not a dir or not writable: ${out_dir@Q}"
+    fi
+    local source="$in_dir/${snake_slug}.sh"
+    if ! [[ -f $source && -r $source ]]; then
+        usage "Error: not a file or not readable: ${source@Q}"
+    fi
 }
 
-process() {
-    # this reads stdin and emits to stdout
-    ruby bin/process-shellcheck-output.rb
+array_includes() {
```

Comment:
You could also do this without hoping your values do not contain some magical separator:

```
array_includes() {
    want="$1"
    shift
    for got; do [[ "$want" = "$got" ]] && return; done
}

array_includes want_file "${files[@]}"
```